src/h5web/visualizations/nexus/utils.test.ts

```diff
@@ -0,0 +1,330 @@
+        attributes: [makeStrAttr('signal', 'dataset')],
+      });
+
+      expect(() => findSignalDataset(group)).toThrow(/to have simple shape/u);
+    });
+
+    it('should throw if signal dataset has no dimensions', () => {
+      const dataset = makeDataset('dataset', makeSimpleShape([]), stringType);
+      const group = makeGroup('group', [dataset], {
+        attributes: [makeStrAttr('signal', 'dataset')],
+      });
+
+      expect(() => findSignalDataset(group)).toThrow(/to have dimensions/u);
+    });
+
+    it('should throw if signal dataset has non-numeric type', () => {
+      const dataset = makeDataset('dataset', makeSimpleShape([1]), stringType);
+      const group = makeGroup('group', [dataset], {
+        attributes: [makeStrAttr('signal', 'dataset')],
+      });
+
+      expect(() => findSignalDataset(group)).toThrow(/to have numeric type/u);
+    });
+  });
+
+  describe('isNxInterpretation', () => {
+    it('should return `true` with known NeXus interpretation', () => {
+      expect(isNxInterpretation('spectrum')).toBe(true);
+    });
+
+    it('should return `false` with invalid or unknown NeXus interpretation', () => {
+      expect(isNxInterpretation(42)).toBe(false);
+      expect(isNxInterpretation('unknown')).toBe(false);
+    });
+  });
+
+  describe('getNxAxes', () => {
+    it('should return array of axes', () => {
+      const axesAttr = makeNxAxesAttr(['X', 'Y']);
+      const group = makeGroup('foo', [], { attributes: [axesAttr] });
+
+      expect(getNxAxes(group)).toEqual(['X', 'Y']);
+    });
+
+    it('should wrap single axis string in array', () => {
+      const axesAttr = makeStrAttr('axes', 'X');
+      const group = makeGroup('foo', [], { attributes: [axesAttr] });
+
+      expect(getNxAxes(group)).toEqual(['X']);
+    });
+
+    it('should return empty array if group has no `axes` attribute', () => {
+      const group = makeGroup('foo');
+      expect(getNxAxes(group)).toEqual([]);
+    });
+
+    it('should throw if `axes` value is neither string nor array', () => {
+      const axesAttr = makeScalarIntAttr('axes', 42);
+      const group = makeGroup('foo', [], { attributes: [axesAttr] });
+
+      expect(() => getNxAxes(group)).toThrow(/array/u);
+    });
+  });
+
+  describe('getDatasetLabel', () => {
+    it('should return value of `long_name` attribute in priority if available', () => {
+      const dataset = makeDataset('foo', scalarShape, intType, {
+        attributes: [
+          makeStrAttr('long_name', 'Foobar'),
+          makeStrAttr('units', 'nm'),
+        ],
+      });
+
+      expect(getDatasetLabel(dataset)).toBe('Foobar');
+    });
+
+    it('should combine dataset name and value of `units` attribute if available', () => {
+      const dataset = makeDataset('foo', scalarShape, intType, {
+        attributes: [makeStrAttr('units', 'nm')],
+      });
+
+      expect(getDatasetLabel(dataset)).toBe('foo (nm)');
+    });
+
+    it('should fall back to dataset name', () => {
```

Comment:
...when other attributes are not string ?

Reply:
👍 I've used should fall back to dataset name when other attributes are absent or not string and added a check for a dataset with no attributes.